src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs

```diff
@@ -118,26 +285,266 @@ public void AddAuditRule_InvalidObjectAuditRule()
+            List<CustomAuditRule> addedRules = ruleCollection.Cast<CustomAuditRule>().ToList();
+
+            Assert.True(addedRules.Contains(customAuditRuleRead));
+            Assert.True(addedRules.Contains(customAuditRuleReadAttribute));
+        }
+
+        [Fact]
+        public void RemoveAccessRule_InvalidObjectAccessRule()
+        {
+            var customObjectSecurity = new CustomDirectoryObjectSecurity();
+            AssertExtensions.Throws<ArgumentNullException>("rule", () => customObjectSecurity.RemoveAccessRule(null));
+        }
+
+        [Fact]
+        public void RemoveRule_AccessControlType_Allow_Succeeds()
+        {
+            var descriptor = new CommonSecurityDescriptor(true, true, string.Empty);
+            var customObjectSecurity = new CustomDirectoryObjectSecurity(descriptor);
+            // read access mask  : 0x01
+            // write access mask : 0x02
+            int readWriteAccessMask = 0x01 | 0x02;
+            var objectTypeGuid = Guid.NewGuid();
+            var identityReference = new NTAccount(@"NT AUTHORITY\SYSTEM");
+
+            var customAccessRuleReadWrite = new CustomAccessRule(
+                identityReference, readWriteAccessMask, true, InheritanceFlags.None,
+                PropagationFlags.None, objectTypeGuid, Guid.NewGuid(), AccessControlType.Allow
+                );
+
+            int writeAccessMask = 0x02;
+            var customAccessRuleWrite = new CustomAccessRule(
+                identityReference, writeAccessMask, true, InheritanceFlags.None,
+                PropagationFlags.None, objectTypeGuid, Guid.NewGuid(), AccessControlType.Allow
+                );
+
+            customObjectSecurity.AddAccessRule(customAccessRuleReadWrite);
+            bool result = customObjectSecurity.RemoveAccessRule(customAccessRuleWrite);
+
+            Assert.Equal(true, result);
+            AuthorizationRuleCollection ruleCollection =
+                customObjectSecurity
+               .GetAccessRules(true, true, typeof(System.Security.Principal.NTAccount));
+
+            Assert.NotNull(ruleCollection);
+            List<CustomAccessRule> existingRules = ruleCollection.Cast<CustomAccessRule>().ToList();
+            int readAccessMask = 0x01;
+
+            Assert.True(
+                existingRules.Any(
+                    x => x.IdentityReference == identityReference &&
+                    x.AccessControlType == customAccessRuleReadWrite.AccessControlType &&
+                    x.AccessMaskValue == readAccessMask
+                    ));
+        }
+
+        [Fact]
+        public void RemoveAccessRule_AccessControlType_Deny_Succeeds()
+        {
+            var descriptor = new CommonSecurityDescriptor(true, true, string.Empty);
+            var customObjectSecurity = new CustomDirectoryObjectSecurity(descriptor);
+
+            // read access mask : 0x01
+            // read attribute access mask : 0x80
+            int readDataAndAttribute = 0x01 | 0x80;
```

Comment:
nit: it'd be nice to have these values in a const and shared between tests if they're not exposed out of Sys.IO.FileSystem.AC